syntax				= "proto3";
option optimize_for = CODE_SIZE;

message Cfg_AIType
{

		uint32 idmonster				= 1;  //怪物id
		uint32 ai_type					= 2;  // ai类型: 0=木桩 1=被动 2=主动
		uint32 idle_randomwalk_ms_min	= 3;  // idle时,随机移动间隔
		uint32 idle_randomwalk_ms_max	= 4;  // idle时,随机移动间隔
		float  idle_randomwalk_step_min = 5;  // idle时,随机移动单步距离
		float  idle_randomwalk_step_max = 6;  // idle时,随机移动单步距离
		float  idle_randomwalk_range	= 7;  //随机移动最大范围,出了范围,走回原始点
		float  attack_pursue_selfrange	= 8;  //最大追击距离,自己超出范围,回到原始点
		float  attack_target_range		= 9;  //最大目标距离,超出后,放弃目标
		float  search_enemy_range		= 10; //最大索敌距离
		float  search_enemy_ms_min		= 11; //索敌间隔
		float  search_enemy_ms_max		= 12; //索敌间隔
		uint32 patrol_wait_ms_min		= 13; //巡逻点到达后的随机等待时间
		uint32 patrol_wait_ms_max		= 14; //巡逻点到达后的随机等待时间
		float  escape_range_min			= 15; //逃离最小距离
		float  escape_range_max			= 16; //逃离最大距离
		bool   can_approach				= 17; //是否会追击
		uint32 attack_wait_ms			= 18; //攻击间隔

		uint32			targetfam_id = 20; //目标模糊选择器ID
		uint32			skillfam_id  = 21; //技能模糊选择器ID
		repeated uint32 skill_id	 = 22; //技能id
		uint32			follow_path	 = 23; //巡逻路径编号
		uint32			script_id	 = 24; //脚本id
		bool			ai_debug	 = 25; //是否打开调试信息输出

}